packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -2071,4 +2071,20 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
 
   @override
   TextRange getWordBoundary(TextPosition position) => paragraph.getWordBoundary(position);
+
+  @override
+  TextSelection? getLocalTextSelection() {
+    if (_textSelectionStart != null && _textSelectionEnd != null) {
+      final int start = math.min(_textSelectionStart!.offset, _textSelectionEnd!.offset);
+      final int end = math.max(_textSelectionStart!.offset, _textSelectionEnd!.offset);
+
+      return TextSelection(baseOffset: start, extentOffset: end);
+    }
+    return null;
+  }
+
+  @override
+  int? getContentLength() {
```

Comment:
What is the use case for this field?

Reply:
In the MultiSelectableSelectionContainerDelegate, (where we actually get our global offsets) getContentLength is used method  to add up the lengths of content from each unselected selectable. This helps to figure out where the text selection's index should begin.